src/collector.ts

```diff
@@ -124,22 +126,35 @@ class NaiveGomodParser {
                         const entry: IKeyValueEntry = new KeyValueEntry(pkgName, { line: 0, column: 0 });
                         entry.value = new Variant(ValueType.String, 'v' + version[0]);
                         entry.value_position = { line: index + 1, column: version.index };
-                        dependencies.push(new Dependency(entry));
-
-                        // Find packages of this module.
-                        goImports.forEach(pckg => {
-                            if (pckg != pkgName && pckg.startsWith(pkgName + "/")) {
-                                const entry: IKeyValueEntry = new KeyValueEntry(pckg, { line: 0, column: 0 });
-                                entry.value = new Variant(ValueType.String, 'v' + version[0]);
-                                entry.value_position = { line: index + 1, column: version.index };
-                                dependencies.push(new Dependency(entry));
-                            }
-                        })
+                        dependencies.push(new Dependency(entry, pkgName));
                     }
                 }
             }
             return dependencies;
         }, []);
+
+        const totalDirectDeps = dependencies.length;
```

Comment:
I'm bit confused about the logic over here.

There are 2 sets,
1. Modules parsed from go.mod
2. Packages identified using `go list`.

Simple logic should be returning all items in #2 with location & mod info from #1. This would also eliminate unnecessary baggage from go.mod. 

Also can't we encode module name with pkg name similar to SA? `pkg/name@mod/name`? This eliminate the changes to Dependency class.

Reply:
Here #2 is output of `go list -f '{{ join .Imports "\\n" }}' ./...`, which simply has list of import and no version / module information. We don't want to take entire -deps output for CA and than add complicated parsing to derived things from go list. In case SA we are doing complicated manifest generation process in go build manifest pacakge. We don't have that choice here. I think current logic is more optimal for CA call.

May be we can bring one small improvement of 
1. First build map of module -> version from go.mod.
2. Loop through imports and build deps using #1 map.

But we cannot avoid looping through go.mod and import list.